pkg/madmin/info-commands.go

```diff
@@ -288,3 +290,51 @@ func (adm *AdminClient) ServerMemUsageInfo() ([]ServerMemUsageInfo, error) {
 
 	return info, nil
 }
+
+// Permission holds the info for encrypt and decrypt
+type Permission struct {
+	Encryption string `json:"encryption,omitempty"`
+	Decryption string `json:"decryption,omitempty"`
+}
+
+// VaultInfo Contains the vault info
+type VaultInfo struct {
+	Endpoint string     `json:"endpoint"` // The vault API endpoint as URL
+	Name     string     `json:"name"`     // The name of the encryption key-ring
+	Type     string     `json:"type"`     // The authentication type
+	Perm     Permission `json:"perm"`
+}
+
+// ServerVaultInfo fetches the vault info
+func (adm *AdminClient) ServerVaultInfo() (VaultInfo, error) {
+	v := url.Values{}
+	v.Set("infoType", string("vault"))
```

Comment:
Why do we need to 'cast' it using the `string` function?

Reply:
removed.